docs/protocol-v3.md

```diff
@@ -0,0 +1,34 @@
+# Integration Protocol v3
+
+This version is basically the same as **v2**, but it improves **remote entities uniqueness** by adding "local address replacement on entity names" at agent level.
+
+
+## Agent support
+
+This integration protocol is supported by New Relic infrastructure agent since version **TBD**.
+
+
+## Name local address replacement
+
+When several remote entities have their `name` based on an endpoint (either ip or hostname), and this name contains a 
+[loopback addresses](https://en.wikipedia.org/wiki/Localhost#Name_resolution), we have 2 issues:
+
+- This localhost value does not provide valuable info without more context.
+- Name could collide with other service being named with a local address.
+
+This happens when:
+
+- Endpoints names are like `localhost:port`.
+- Ports tend be the same for a given service. Ie: 3306 for Mysql.
+
+
```

Comment:
I would add a paragraph explaining what the agent does, something in the lines of:

> In this case, when the agent receives an entity key from a protocol v3 payload containing a loopback reference, it will replace such reference with ...